tasks/snyk-code-test.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: snyk-code-test
+spec:
+  description: >-
+    Static code security test with snyk
+  results:
+    - description: Test output
+      name: HACBS_TEST_OUTPUT
+  params:
+    - name: shared-secret
+      default: test-team-snyk
+    - name: args
+      type: string
+      description: extra args needs to append
+      default: "--all-projects --exclude=test*,vendor,deps"
+  volumes:
+    - name: snyk-secret
+      csi:
+        readOnly: true
+        driver: csi.sharedresource.openshift.io
+        volumeAttributes:
+          sharedSecret: $(params.shared-secret)
+  steps:
+    - name: snyk-code-test
+      image: quay.io/redhat-appstudio/hacbs-test:feature-sast
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      volumeMounts:
+        - name: snyk-secret
+          mountPath: "/etc/secrets"
+          readOnly: true
+      script: |
+        #!/usr/bin/env bash
+        SNYK_TOKEN="$(cat /etc/secrets/snyk_token)"
+        if [[ -e $SNYK_TOKEN ]]; then
+          echo "SNYK_TOKEN is empty and a secret 'test-team-snyk' which includes 'snyk_token' is needed to be created in test_team namespace for it" 
+          exit 1
+        fi
+        export SNYK_TOKEN
+        pushd $(workspaces.workspace.path)
```

Comment:
workingDir is not used, so it can be set to workingDir instead of pushd.

Reply:
Updated